lib/src/rules/use_build_context_synchronously.dart

```diff
@@ -38,6 +40,20 @@ void onButtonTapped(BuildContext context) {
 }
 ```
 
+**GOOD:**
+```dart
+void onButtonTapped() async {
+  final navigator = Navigator.of(context);
+  await Future.delayed(const Duration(seconds: 1));
+
+  navigator.pop();
+}
+```
+
+**EXCEPTION:**
+In a `StatefulWidget` it is possible to check the `State`'s `mounted` property, after an
+asynchronous gap and before using `BuildContext`.
```

Comment:
This does not match the example below where `mounted` is called on the `BuildContext` and not the `State`.

This statement is also confusing. With the latest version of Flutter, `mounted` is a property of `BuildContext` and that one should be used over the `State` property as it is also available in StatelessWidgets.

Please remove this and replace this with a note that starting with Flutter version xxx, one can also check the mounted property on the BuildContext. We could then also have a section that talks about older versions of Flutter where this was only available on the State. But we should make clear that going forward `BuildContext.mounted` is the right way to do these checks.